bundle/jsky.app.ot/src/main/java/jsky/app/ot/editor/template/EdTemplateGroup.java

```diff
@@ -379,22 +372,29 @@ private void addNewParameters(Function1<Option<TemplateParameters>, TemplatePara
         // prototype.
         final Option<TemplateParameters> proto = (sel.length == 1) ?
                 new Some<TemplateParameters>((TemplateParameters) templateGroup.getTemplateParameters().get(sel[0]).getDataObject()) :
-                None.<TemplateParameters>instance();
+                None.instance();
 
         // Make a new template parameters object with the provided constructor.
-        final ISPTemplateParameters newParams;
-        try {
-            newParams = SPDB.get().getFactory().createTemplateParameters(program, null);
-            newParams.setDataObject(cons.apply(proto));
-            templateGroup.addTemplateParameters(where, newParams);
-            SwingUtilities.invokeLater(new Runnable() {
-                @Override public void run() {
-                    paramTable.getSelectionModel().setSelectionInterval(where, where);
-                }
-            });
-        } catch (SPException ex) {
-            DialogUtil.error(ex);
-        }
+        final ImList<TemplateParameters>      dobs = cons.apply(proto);
+        final ImList<ISPTemplateParameters> shells = dobs.map(dob -> {
+            final ISPTemplateParameters shell;
+            try {
+                shell = SPDB.get().getFactory().createTemplateParameters(program, null);
+            } catch (SPException ex) {
+                throw new RuntimeException(ex);
+            }
+            shell.setDataObject(dob);
+            return shell;
```

Comment:
This isn't really kosher (especially throwing the exception) but since this is Java, I'm inclined to leave it.  If you squint, it's just transforming the data objects to a shell with the data object set.